massa-execution-worker/src/tests/scenarios_mandatories.rs

```diff
@@ -1,3115 +1,2484 @@
-    }
+#[test]
+fn validate_address() {
+    // setup the period duration
+    let exec_cfg = ExecutionConfig {
+        t0: MassaTime::from_millis(100),
+        cursor_delay: MassaTime::from_millis(0),
+        ..ExecutionConfig::default()
+    };
+    let block_producer = KeyPair::generate(0).unwrap();
+    let keypair = KeyPair::from_str(TEST_SK_1).unwrap();
+
+    let mut foreign_controllers = ExecutionForeignControllers::new_with_mocks();
+    selector_boilerplate(
+        &mut foreign_controllers.selector_controller,
+        &block_producer,
+    );
+    let mut universe = ExecutionTestUniverse::new(foreign_controllers, exec_cfg.clone());
+
+    // load bytecode
+    // you can check the source code of the following wasm file in massa-unit-tests-src
+    let bytecode = include_bytes!("./wasm/validate_address.wasm");
+
+    // create the block containing the erroneous smart contract execution operation
+    let operation =
+        ExecutionTestUniverse::create_execute_sc_operation(&keypair, bytecode, BTreeMap::default())
+            .unwrap();
+    universe.storage.store_operations(vec![operation.clone()]);
+    let block = ExecutionTestUniverse::create_block(
+        &block_producer,
+        Slot::new(1, 0),
+        vec![operation],
+        vec![],
+        vec![],
+    );
+    // store the block in storage
+    universe.storage.store_block(block.clone());
+
+    // set our block as a final block
+    let mut finalized_blocks: HashMap<Slot, BlockId> = Default::default();
+    finalized_blocks.insert(block.content.header.content.slot, block.id);
+    let block_store = vec![(
+        block.id,
+        ExecutionBlockMetadata {
+            storage: Some(universe.storage.clone()),
+            same_thread_parent_creator: Some(Address::from_public_key(&keypair.get_public_key())),
+        },
+    )]
+    .into_iter()
+    .collect();
+    universe.module_controller.update_blockclique_status(
+        finalized_blocks,
+        Default::default(),
+        block_store,
+    );
+    std::thread::sleep(
+        exec_cfg
+            .t0
+            .saturating_add(MassaTime::from_millis(50))
+            .into(),
+    );
+
+    let events = universe
+        .module_controller
+        .get_filtered_sc_output_event(EventFilter::default());
+    // match the events
+    assert_eq!(events.len(), 2);
+    assert!(
+        events[0].data.ends_with("true"),
+        "Expected 'true': {:?}",
+        events[0].data
+    );
+    assert!(
+        events[1].data.ends_with("false"),
+        "Expected 'false': {:?}",
+        events[1].data
+    );
+
+    assert_eq!(
+        universe
+            .final_state
+            .read()
+            .ledger
+            .get_balance(&Address::from_public_key(&keypair.get_public_key()))
+            .unwrap(),
+        Amount::from_str("300000")
+            .unwrap()
+            // Gas fee
+            .saturating_sub(Amount::const_init(10, 0))
+    );
+}
 
-    #[test]
-    fn test_take_batch() {
```

Comment:
This is covered by the pool tests now, right?

Reply:
Yes